mipengine/controller/cleaner.py

```diff
@@ -39,63 +54,115 @@ class Config:
+                                entry.context_id
+                            )
+                            self._logger.debug(
+                                f"Creating file with {entry.context_id=}"
+                            )
+                            self._cleanup_files_processor.create_file_from_cleanup_entry(
+                                entry
+                            )
+
+                        else:
+                            self._logger.debug(
+                                f"Cleanup for {entry.context_id=} complete. Deleting file"
+                            )
+                            self._cleanup_files_processor.delete_file_by_context_id(
+                                entry.context_id
+                            )
+
+            except Exception:
+                self._logger.error(traceback.format_exc())
 
-    def _process_contextid(self, context_id: str, status: dict):
-        if status["nodes"]:
-            if status["released"] or self._is_expired(status["timestamp"]):
-                self._cleanup_nodes_list(
-                    context_id=context_id, node_ids=status["nodes"]
-                )
-        else:
-            self._remove_contextid_from_cleanup(context_id=context_id)
-
-    def _is_expired(self, timestamp: str):
-        now = datetime.now(timezone.utc)
-        timestamp = datetime.fromisoformat(timestamp)
-        time_elapsed = now - timestamp
-        return (
-            time_elapsed.seconds > controller_config.cleanup.contextid_release_timelimit
-        )
+            finally:
+                time.sleep(self._cleanup_interval)
 
-    def _cleanup_nodes_list(self, context_id: str, node_ids: List[str]):
+    def _exec_context_id_cleanup(
+        self, context_id: str, node_ids: [str]
+    ) -> List[str]:  # returns failed node_ids
+        node_task_handlers_to_async_results = {}
         for node_id in node_ids:
             try:
                 node_info = self._get_node_info_by_id(node_id)
-                task_handler = _create_node_task_handler(node_info)
-                task_handler.clean_up(
-                    request_id=CLEANER_REQUEST_ID,
-                    context_id=context_id,
-                )
-                self._remove_nodeid_from_cleanup(context_id=context_id, node_id=node_id)
+            except Exception:
                 self._logger.debug(
-                    f"clean_up task succeeded for {node_id=} for {context_id=}"
+                    f"Could not get node info for {node_id=}. The node is "
+                    "most likely offline"
+                )
+                continue
+            task_handler = _get_node_task_handler(node_info)
+
+            node_task_handlers_to_async_results[
+                task_handler
+            ] = task_handler.queue_cleanup(
+                request_id=CLEANER_REQUEST_ID,
+                context_id=context_id,
+            )
```

Comment:
yes it can fail. It can fail if the node's rabbitmq is unreachable which most likely will be caught in the previous call 
```
node_info = self._get_node_info_by_id(node_id)
```
but even if the node goes down after that so the 
```
task_handler.queue_cleanup(...)
```
fails the `_exec_context_id_cleanup` will just return and will retry later. I prefer that than put more try blocks in there

Reply:
The scenario where it will fail is the following:
1. The node is up and it's added in the NLA registries.
2. The node goes down.
3. The cleanup gets the node id from the registry.
4. The cleanup tries to queue and fails.
5. The node id is not added in the failed node ids.